services/red/src/api/Service/RatesService.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using artiso.AdsdHotel.ITOps.Communication.Abstraction;
+using artiso.AdsdHotel.ITOps.Communication.Abstraction.NServiceBus;
+using artiso.AdsdHotel.Red.Contracts;
+using artiso.AdsdHotel.Red.Events;
+using artiso.AdsdHotel.Red.Persistence;
+using artiso.AdsdHotel.Red.Persistence.Entities;
+using Grpc.Core;
+using ConfirmationDetails = artiso.AdsdHotel.Red.Contracts.ConfirmationDetails;
+
+namespace artiso.AdsdHotel.Red.Api.Service
+{
+    public sealed class RatesService : Rates.RatesBase
+    {
+
+        private readonly IRoomPriceService _roomPriceService;
+        private readonly IChannel _channel;
```

Comment:
I think that is okay here, because the communication from ambassador to the service is via GRPC, hence the service has no `IMessageHandlerContext` and has to use a sending endpoint.

Reply:
in red service the api is the only project which uses nservicebus. Communication from ambassador to the api is done with grpc